src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,262 @@
+        user1.setLogin("12345");
+        user1.setPassword("12345");
+
+        User user2 = new User();
+        user2.setAge(18);
+        user2.setLogin("123456");
+        user2.setPassword("123456");
+
+        User user3 = new User();
+        user3.setAge(19);
+        user3.setLogin("1234567");
+        user3.setPassword("1234567");
+
+        User user4 = new User();
+        user4.setAge(20);
+        user4.setLogin("12345678");
+        user4.setPassword("12345678");
+
+        storageDao.add(user1);
+        storageDao.add(user2);
+        storageDao.add(user3);
+        storageDao.add(user4);
+
+        users.add(user1);
+        users.add(user2);
+        users.add(user3);
+        users.add(user4);
+    }
+
+    @Test
+    void register_user_OK() {
+        User user = new User();
+        user.setId(5L);
+        user.setAge(19);
+        user.setLogin("123456789");
+        user.setPassword("123456789");
+
+        User expected = user;
+        User actual = registrationService.register(user);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_login_OK() {
+        User user = new User();
+        user.setId(5L);
+        user.setAge(19);
+        user.setLogin("012345");
+        user.setPassword("012345");
+
+        Integer expected = user.getLogin().length();
+        Integer actual = registrationService.register(user).getLogin().length();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_password_OK() {
+        User user = new User();
+        user.setId(5L);
+        user.setAge(19);
+        user.setLogin("012345");
+        user.setPassword("012345");
+
+        Integer expected = user.getPassword().length();
+        Integer actual = registrationService.register(user).getPassword().length();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_age_18_OK() {
+        User user = new User();
+        user.setId(5L);
+        user.setAge(18);
+        user.setLogin("012345");
+        user.setPassword("012345");
+
+        Integer expected = user.getAge();
+        Integer actual = registrationService.register(user).getAge();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_age_25_OK() {
+        User user = new User();
+        user.setId(5L);
+        user.setAge(25);
+        user.setLogin("012345");
+        user.setPassword("012345");
+
+        Integer expected = user.getAge();
+        Integer actual = registrationService.register(user).getAge();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_null_OK() {
+        User user = new User();
+        user.setId(5L);
+        user.setAge(18);
+        user.setLogin("012345");
+        user.setPassword("012345");
+
+        User expected = user;
+        User actual = registrationService.register(user);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_user_notOK() {

```

Comment:
What is 'not ok'?

Reply:
it is uncorrect work. It was written by example in Readme.md - register_nullAge_notOk